docker/start.py

```diff
@@ -59,6 +59,13 @@ def generate_secrets(environ, secrets):
         if not os.path.exists("/compiled"): os.mkdir("/compiled")
 
         config_path = "/compiled/homeserver.yaml"
+        
+        # Convert SYNAPSE_NO_TLS to boolean if exists
+        if "SYNAPSE_NO_TLS" in environ:
+            if str.lower(environ["SYNAPSE_NO_TLS"]) in ("true", "on", "1", "yes"):
+                environ["SYNAPSE_NO_TLS"] = True
+            else:
+                environ["SYNAPSE_NO_TLS"] = False
```

Comment:
I wonder if we should throw an error if they put in something that isn't "false", "0", etc to avoid people getting confused if they accidentally typo e.g. "ture" or something.

In fact I think we use `from distutils.util import strtobool` somewhere else that does this for us?

Reply:
not sure if you can rely on `distutils` being available in the env for this script?

+1 to barfing on unrecognised values though.

Reply:
Oh, you're right ofc, I got confused and thought we were in `synapse.config` :man_facepalming:

Reply:
I haven't replied for a while but I do plan on enhancing this somewhat soon.

Reply:
I think https://github.com/matrix-org/synapse/pull/5005/commits/182e46ce71e17b7f540967b3149a8952040b8694 resolves this...?